File: C++/1220_CountVowelsPermutation.cpp
class Solution {
public:
    int countVowelPermutation(int n) {
        const int mod = pow(10, 9)+7;
        long a = 1, e = 1, i = 1, o = 1, u = 1;
        for (int z = 1; z < n; z++){
            long aa = a;
            long ee = e;
            long ii = i;
            long oo = o;
            long uu = u;
            a = (ee + ii + uu) % mod;
            e = (aa + ii) % mod;
            i = (ee + oo) % mod;
            o = ii;
            u = (ii + oo) % mod;
        }
        return (a+e+i+o+u) % mod;
    }
};

File: 1220_CountVowelsPermutation.py
class Solution:
    def countVowelPermutation(self, n: int) -> int:
        vowel = {'a': 1, 'e': 1,'i':1,'o':1,'u':1}
        for i in range(1,n):
            a = vowel['a'] 
            e = vowel['e']
            i = vowel['i']
            o = vowel['o']
            u = vowel['u']
            vowel['a'] = e + i + u
            vowel['e'] = a + i
            vowel['i'] = e + o
            vowel['o'] = i
            vowel['u'] = i + o
        return (vowel['a'] + vowel['e'] + vowel['i'] + vowel['o'] + vowel['u']) % (10**9 + 7)